src/panels/calendar/ha-panel-calendar.js

```diff
@@ -0,0 +1,227 @@
+        <app-header slot="header" fixed>
+          <app-toolbar>
+            <ha-menu-button narrow='[[narrow]]' show-menu='[[showMenu]]'></ha-menu-button>
+            <div main-title>[[localize('panel.calendar')]]</div>
+          </app-toolbar>
+        </app-header>
+
+        <div class="flex content">
+          <div id="calendars" class="layout vertical wrap">
+            <paper-card heading="Calendars">
+              <div class="all_calendars">
+                <paper-checkbox id="all_calendars" on-change="checkAll" checked>All calendars</paper-checkbox>
+              </div>
+              <paper-listbox id="calendar_list" multi on-selected-items-changed="_fetchData" selected-values="{{selectedCalendars}}" attr-for-selected="item-name">
+                <template is="dom-repeat" items="[[calendars]]">
+                  <paper-item item-name="[[item.entity_id]]">
+                    <span class="calendar_color" style$="background-color: [[item.color]]"></span>
+                    <span class="calendar_color_spacer"></span>
+                    [[item.name]]
+                  </paper-item>
+                </template>
+              </paper-listbox>
+            </paper-card>
+          </div>
+          <div class="flex layout horizontal wrap">
+            <ha-big-calendar
+              default-date="[[currentDate]]"
+              default-view="[[currentView]]"
+              on-navigate='handleNavigate'
+              on-view='handleNavigate'
+              events="[[events]]">
+            </ha-big-calendar>
+          </div>
+        </div>
+      </app-header-layout>`;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._fetchData = this._fetchData.bind(this);
+    // TODO implement calendar_updated event
+    // this.hass.connection.subscribeEvents(this._fetchData, 'calendar_updated')
+    //  .then(function (unsub) { this._unsubEvents = unsub; }.bind(this));
+    this._fetchCalendar();
+  }
+
+  _fetchCalendar() {
+    // Fetch calendar list
+    this.hass.callApi('get', 'calendars')
+      .then((result) => {
+        this.calendars = result;
+      });
+  }
+
+  _fetchData() {
+    var dates = this._getDateRange();
+    var start = dates[0];
+    var end = dates[1];
+    // Fetch calendar list
+    this._fetchCalendar();
+    // Fetch events for selected calendar
+    const params = encodeURI(`?start=${start}&end=${end}`);
+    const calls = this.selectedCalendars.map(cal => this.hass.callApi('get', `calendar/${cal}${params}`));
+    Promise.all(calls).then((results) => {
+      var tmpEvents = [];
+
+      results.map((res) => {
+        res.map((ev) => {
+          ev.start = new Date(ev.start);
+          if (ev.end) {
+            ev.end = new Date(ev.end);
+          } else {
+            ev.end = null;
+          }
+          tmpEvents.push(ev);
+          return null;
+        });
+        return null;
+      });
+      this.events = tmpEvents;
+    });
+  }
+
+  _getDateRange() {
```

Comment:
This can be changed in a future PR.

It's weird that you calculate this yourself. Each view object in Big Calendar implements a `navigate` and `range` helper method. [Example](https://github.com/intljusticemission/react-big-calendar/blob/master/src/Week.js#L23)

Reply:
And you can pass down the view object (or calculate the exact info that is needed by the parent component) by getting the View object inside the navigate method using `BigCalendar.Views[viewName]`

Reply:
We should ask the view object for the range. They all have it:

 - [Week](https://github.com/intljusticemission/react-big-calendar/blob/master/src/Week.js#L36)
 - [Day](https://github.com/intljusticemission/react-big-calendar/blob/master/src/Day.js#L22)
etc

Reply:
Hum, I don't understand how I can do that :/
Which view object are you speaking about ?
I tried:
```
import Week from 'react-big-calendar';
```
But `Week.range` doesn't exists.

What am I missing ?!

Reply:
`import views from 'react-big-calendar/lib/Views.js'`

Reply:
If you don't specify specific names on the left side (using `{ name }`), you are going to import the default export. The default export for `'react-big-calendar'` is specified as `main` in `package.json`. That one does not know about views. So we have to peak into a specific file, the `Views.js` file inside `lib`

Reply:
Anyway, this is for a future PR.